portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -80,6 +104,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
-    response.sendRedirect("/comments.html");
+    String url = "/comments.html?numComments=" + numDisplay + "&commentSort=" + sortOrder;
+
+    response.sendRedirect(url);
+  }
+
+  /**
+   * Gets the sort order from the request header, and parses it accordingly.
+   * @return an integer indicating the sort order.
+   */
+  private int getSortOrder(HttpServletRequest request) {
+    String data = request.getHeader("commentSort");
+    if ("timeAscending".equals(data)) {
+      return TIME_ASCENDING;
+    } else if ("timeDescending".equals(data)) {
+      return TIME_DESCENDING;
+    } else {
+      return -1;
```

Comment:
It seems weird that you use constants for 2 of these values but not for the error value.  What if in the future someone decided that it would be easier to have TIME_ASCENDING = -1 and TIME_DESCENDING = 1.  This would break your code.  You probably want to create a constant for the error condition that you could refer to later.